swan-lake/featured-use-cases/work-with-data-using-queries-in-ballerina.md

```diff
@@ -434,4 +434,268 @@ Countries with highest deaths:[["India",980976],["USA",880976],["Afghanistan",73
 Countries with number of Recovered patients:[["Afghanistan",146084],["USA",43892277],["India",33892279]]
 Found erroneous entries for countries: ["Sri Lanka","India"]
 ```
-   
\ No newline at end of file
+## Group COVID-19 data by continent
+
+### Create the groupDeathsByContinent function
+
+To difine a function to group the data by continent and get the total number of deaths by continent, add the code below to the `main.bal` file.
+
+```ballerina
+public function groupDeathsByContinent(table<CovidEntry> dataTable) returns record {|string continent; decimal deaths;|}[] {
+    return from var {continent, deaths} in dataTable
+        group by continent
+        select {continent, deaths: sum(deaths)};
+}
+```
+
+In this code,
+1. You use `group by` clause to group the input.
+2. Since the `grouping-key` is `continent` the input table is grouped by `continent` in the function `groupDeathsByContinent`.
```

Comment:
`Have we introduced what `grouping-key` is? If not, users might be confused no knowing what it is.